pkg/uefivars/boot/boot.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2015-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package boot
+
+import (
+	"encoding/binary"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/uefivars"
+)
+
+type BootLabel string
+
+type BootEntry struct {
+	Device   string
+	PartNum  uint
+	Label    BootLabel
+	AbsPath  string
+	Args     string
+	autoBoot bool
+}
+
+const (
+	BootUuid = "8be4df61-93ca-11d2-aa0d-00e098032b8c"
+)
+
+//A boot entry. Will have the name BootXXXX where XXXX is hexadecimal
+type BootEntryVar struct {
+	Number uint16 //from the var name
+	EfiLoadOption
+}
+
+/* EfiLoadOption defines the data struct used for vars such as BootXXXX.
+As defined in UEFI spec v2.8A:
+    typedef struct _EFI_LOAD_OPTION {
+        UINT32 Attributes;
+        UINT16 FilePathListLength;
+        // CHAR16 Description[];
+        // EFI_DEVICE_PATH_PROTOCOL FilePathList[];
+        // UINT8 OptionalData[];
+    } EFI_LOAD_OPTION;
+*/
+type EfiLoadOption struct {
+	Attributes         uint32
+	FilePathListLength uint16
+	Description        string
+	FilePathList       EfiDevicePathProtocolList
+	OptionalData       []byte
+}
+type BootEntryVars []*BootEntryVar
+
+// Gets BootXXXX var, if it exists
+func ReadBootVar(num uint16) (b *BootEntryVar) {
+	v, err := uefivars.ReadVar(BootUuid, fmt.Sprintf("Boot%04X", num))
+	if err != nil {
+		log.Printf("reading var Boot%04X: %s", num, err)
+		return
+	}
+	return BootVar(v)
+}
+
+// Reads BootCurrent, and from there gets the BootXXXX var referenced.
+func ReadCurrentBootVar() (b *BootEntryVar) {
+	curr := ReadBootCurrent()
+	if curr == nil {
+		return nil
+	}
+	return ReadBootVar(curr.Current)
+}
+
+func (b BootEntryVar) String() string {
+	opts, err := uefivars.DecodeUTF16(b.OptionalData)
+	if err != nil {
+		opts = string(b.OptionalData)
+	}
+	//could decode uefi path... worth it?
+	return fmt.Sprintf("Boot%04X: attrs=0x%x, desc=%q, path=%s, opts=%x", b.Number, b.Attributes, b.Description, b.FilePathList.String(), opts)
+}
+
+func (b BootEntryVar) Remove() error {
+	return RemoveBootEntry(b.Number)
+}
+
+//returns list of boot entries (BootXXXX)
+//note that BootCurrent, BootOptionSupport, BootNext, BootOrder, etc do not count as boot entries.
+func AllBootEntryVars() BootEntryVars {
+	//BootEntries() is somewhat redundant, but parses the vars into BootEntryVar{}
+	return BootEntries(uefivars.ReadVars(BootEntryFilter))
+}
+
+//Return all boot-related uefi vars
```

Comment:
can you describe in comment what the criteria for "boot-related" is? Also, try to conform to Go comment standards, starting the comment with the name of the function:

`// AllBootVars returns UEFI vars that either begin with the name "Boot" or match the BootUUID.`

Reply:
sure